Src/ModelLoadCMO.cpp

```diff
@@ -589,29 +599,112 @@ std::unique_ptr<Model> DirectX::Model::CreateFromCMO(
 
                 auto boneName = reinterpret_cast<const wchar_t*>(meshData + usedSize);
 
-                usedSize += sizeof(wchar_t)*(*nName);
+                usedSize += sizeof(wchar_t) * (*nName);
                 if (dataSize < usedSize)
                     throw std::runtime_error("End of file");
 
-                // TODO - What to do with bone name?
-                boneName;
+                bones[j].name = boneName;
 
                 // Bone settings
-                auto bones = reinterpret_cast<const VSD3DStarter::Bone*>(meshData + usedSize);
+                auto cmobones = reinterpret_cast<const VSD3DStarter::Bone*>(meshData + usedSize);
                 usedSize += sizeof(VSD3DStarter::Bone);
                 if (dataSize < usedSize)
                     throw std::runtime_error("End of file");
 
-                // TODO - What to do with bone data?
-                bones;
+                transforms[j] = XMLoadFloat4x4(&cmobones->LocalTransform);
+                invTransforms[j] = XMLoadFloat4x4(&cmobones->InvBindPos);
+
+                if (cmobones->ParentIndex < 0)
+                {
+                    if (!j)
```

Comment:
This is a very C-way of doing this check... is there a more C++ way? Otherwise, I think it reads better as `if (j > 0)`

Reply:
Actually ``j == 0``, but sure :)